common/mails/courses/index.ts

```diff
@@ -203,23 +205,57 @@ async function getCourseStartDate(subcourseId: number) {
     return firstLecture;
 }
 
-export async function sendPupilCourseSuggestion(course: Course | Prisma.course, subcourse: Subcourse | Prisma.subcourse) {
+export async function sendPupilCourseSuggestion(course: Course | Prisma.course, subcourse: Subcourse | Prisma.subcourse, actionId: ActionID) {
     const minGrade = subcourse.minGrade;
     const maxGrade = subcourse.maxGrade;
     const courseStartDate = await getCourseStartDate(subcourse.id);
+    const courseCapacity = await getCourseCapacity(subcourse);
+    const { alreadyPromoted, publishedAt } = subcourse;
+    const courseSubject = course.subject;
     const grades = [];
 
+    if (!courseStartDate) {
+        throw new Error(`Cannot send course suggestion mail for subcourse with ID ${subcourse.id}, because that course has no specified first lecture`);
+    }
+
     for (let grade = minGrade; grade <= maxGrade; grade++) {
         grades.push(`${grade}. Klasse`);
     }
 
-    // TODO filter pupils
     const pupils = await prisma.pupil.findMany({
         where: { active: true, isParticipant: true, grade: { in: grades } },
     });
 
-    for (let pupil of pupils) {
-        await Notification.actionTaken(pupil, 'instructor_subcourse_published', {
+    const getDaysDifference = (date: string): number => {
+        const today = new Date().getTime();
+        const published = new Date(date).getTime();
+        const diffInMs = today - published;
+        if (!diffInMs) {
+            return;
+        }
+        const diffInSec = diffInMs / 1000;
+        const diffInMin = diffInSec / 60;
+        const diffInHours = diffInMin / 60;
+        const diffInDays = diffInHours / 24;
+
+        return diffInDays;
+    };
+
+    const isPromotionValid = (publishedAt: string, capacity: number, alreadyPromoted: boolean): boolean => {
+        let daysDiff: number | string;
+        if (publishedAt === null) {
+            daysDiff = publishedAt;
+        } else {
+            daysDiff = getDaysDifference(publishedAt);
+        }
```

Comment:
```suggestion
        const daysDiff: number | null = publishedAt ? getDaysDifference(publishedAt) : null;
```
This should do the job I guess. 👍🏼